### ISSUE TITLE
issues with functions?

### ISSUE THREAD

**Issue by boriwo** [Knetic/govaluate#28]
I wrote this function:
```
	functions := map[string]govaluate.ExpressionFunction{
		"now": func(args ...interface{}) (interface{}, error) {
			return float64(time.Now().UnixNano() / 1e6), nil
		},
	}
```
Executing `1+now()` works but `now()+1` yields "Unable to plan token kind: MODIFIER".

More complex arithmetic expressions including now() trigger other errors.

**Comment by Knetic** [comment]
Functions with empty parens were a bit tricky to implement. I had thought I had good enough tests to make sure that it worked, but evidently I missed a spot. 

Ought to be fixed as of 17f6df82adc63052b0353f5ad17cc7ff19675a62, seems like I just returned the wrong thing from the codepath which handles empty parens. Sorry for the inconvenience!

**Comment by Knetic** [comment]
Woops, didn't mean to close. I'll let you verify that it works in your cases, first.

**Comment by boriwo** [comment]
Thank you for the quick fix! I confirmed that `now()+1` does indeed work now.

But there are other examples that still look funny:

`(now()-1) > 3` for example returns a number instead of a boolean.

**Comment by Knetic** [comment]
This actually uncovered a bug with the parsing phase and minus sign that I'm surprised I hadn't found before now. This case ought to be fixed as of 89c296b421d2a51e926ebdbe6f10480ead31f5b3, and for good measure I've added a battery of tests around how functions should work with various operators.

Sorry for the churn on this one! Let me know if your cases work with that fix.

**Comment by boriwo** [comment]
My test cases all work now! Thank you!
